ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.pde.internal.core.bnd;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IAdapterFactory;
+import org.osgi.service.component.annotations.Component;
+
+import aQute.bnd.build.Project;
+
+@Component(service = IAdapterFactory.class, property = {
+		IAdapterFactory.SERVICE_PROPERTY_ADAPTABLE_CLASS + "=org.eclipse.core.resources.IProject",
+		IAdapterFactory.SERVICE_PROPERTY_ADAPTABLE_CLASS + "=aQute.bnd.build.Project" })
+public class PdeBndAdapter implements IAdapterFactory {
+
+	@Override
+	public <T> T getAdapter(Object adaptableObject, Class<T> adapterType) {
+		if (adapterType == Project.class) {
+			if (adaptableObject instanceof IProject project) {
+				try {
+					return adapterType.cast(BndProjectManager.getBndProject(project).orElse(null));
+				} catch (Exception e) {
+					// can't adapt then...
```

Comment:
Should this be logged at least?

Reply:
The `Exception` is sadly a burden from the BND project that uses throws Exception all the time :-\
In this case it actually *should* never happen but if it does something is wrong and caller can just give up as we can't decide whats going wrong.